src/libxtp/threecenter_gwbse.cc

```diff
@@ -101,11 +108,14 @@ void TCMatrix_gwbse::Fill(const AOBasis& gwbasis, const AOBasis& dftbasis,
   // If cuda is enabled the dft orbitals are sent first to the cuda gpu
   // and memory in the cuda gpu is allocated for the intermediate matrices
 #if defined(USE_CUDA)
-  std::array<CudaMatrix, 2> cuda_matrices = SendDFTMatricesToGPU(dft_orbitals);
+  CudaPipeline cuda_pip;
+  std::array<CudaMatrix, 2> cuda_matrices =
+      SendDFTMatricesToGPU(dft_orbitals, cuda_pip);
   std::array<CudaMatrix, 3> cuda_inter_matrices =
       CreateIntermediateCudaMatrices(dft_orbitals.rows());
```

Comment:
would you make it consistent then and do the same for `CreateIntermediateCudaMatrices`

Reply:
The `CreateIntermediateCudaMatrices` does not need a `CudaPipeline` object